### PR TITLE
Add a solution for search ranking test

### PR DESCRIPTION
**Pull request by ChunLi0819**

### CODE REVIEW

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        /// </summary>
+        [TestMethod]
+        public void BVT_S01_TC01_CanFindGraphSite()
+        {
+            int lowestRanking = Int32.Parse(ConfigurationManager.AppSettings["LowestRanking"]);
+            int ranking = 0;
+            bool isFounded = false;
+            
+            WebDriver.Navigate().GoToUrl("http://" + searchEngine + ".com");
+            List<SearchedResult> searchedResults = Search("Microsoft Graph");
```
Comment (JingyuShao) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: the search keyword should be an enum , better to be configurable in app.config

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            foreach (SearchedResult result in searchedResults)
+            {
+                if (ranking >= lowestRanking) break;
+                if (result.Name.Contains("Microsoft Graph"))
+                {
+                    isFounded = true;
```
Comment (JingyuShao) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: the logic should be judge whether the result.DetailLink contains microsoft.graph.io. if so, isFounded = true. No need to go to the link.

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            {
+                Assert.IsTrue(WebDriver.Title.Contains("Microsoft Graph"), "The result at position {0} should be Microsoft Graph production site", ranking + 1);
+            }
+        }
+
+        /// <summary>
+        /// Verify whether dev.office.com site can be found on search engine
+        /// </summary>
+        [TestMethod]
+        public void BVT_S01_TC02_CanFindGraphSite()
```
Comment (JingyuShao) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: both case same name?

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                {
+                    isFounded = true;
+                    WebDriver.Navigate().GoToUrl(result.DetailLink);
+                    break;
+                }
+                ranking++;
+            }
+            if (!isFounded)
+            {
+                Assert.Fail("Cannot find the production site in top {0}", lowestRanking);
```
Comment (JingyuShao) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: please also add the search engine name and timestamp of running this case.

**Comment by JingyuShao** [comment]
Can you also test run on IE? I run failed for both cases on IE

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs (hunk 2 above)
Comment (ChunLi0819) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: Updated

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs (hunk 1 above)
Comment (ChunLi0819) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: Add an enum for search keywords. Enum value's Description attribute must be a constant value, so cannot set Description as an app setting property value

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs (hunk 4 above)
Comment (ChunLi0819) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: Updated

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs (hunk 3 above)
Comment (ChunLi0819) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: Updated

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs (hunk 1 above)
Comment (JingyuShao) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: Please add these to the enum for graph: Microsoft Graph, Microsoft Graph API, Graph Microsoft
And these for dev.office.com: Office Dev Center, dev office

File: SearchEngineTest/SearchEngineTest/SearchEngineTest.cs (hunk 4 above)
Comment (JingyuShao) [inline SearchEngineTest/SearchEngineTest/SearchEngineTest.cs]: not the time of running the case, but the timestap. Like at which day, what time I run this case. The reason behind is search engine result may be different when you search at different time.

**Pull request merged**